app/scripts/lib/rpc-method-middleware/handlers/switch-ethereum-chain.js

```diff
@@ -13,21 +14,26 @@ import {
   isPrefixedFormattedHexString,
   isSafeChainId,
 } from '../../../../../shared/modules/network.utils';
+import { PermissionNames } from '../../../controllers/permissions';
+import { CaveatTypes } from '../../../../../shared/constants/permissions';
 
 const switchEthereumChain = {
   methodNames: [MESSAGE_TYPE.SWITCH_ETHEREUM_CHAIN],
   implementation: switchEthereumChainHandler,
   hookNames: {
-    getCurrentChainId: true,
     findNetworkConfigurationBy: true,
-    findNetworkClientIdByChainId: true,
     setNetworkClientIdForDomain: true,
-    setProviderType: true,
     setActiveNetwork: true,
-    requestUserApproval: true,
     getNetworkConfigurations: true,
-    getProviderConfig: true,
     hasPermissions: true,
+    hasPermission: true,
+    getPermissionsForOrigin: true,
+    getCaveat: true,
+    requestSwitchNetworkPermission: true,
+    getCurrentChainIdForDomain: true,
+    // old hooks no longer used post chain permissioning:
```

Comment:
They're already no longer used or they _will_ no longer be used at some point in the future?

Reply:
the latter. But removed the comment here: https://github.com/MetaMask/metamask-extension/pull/24415/commits/465ae6bbfdd4a69ef07cf122bac5fb214a4ecd98